### PR TITLE
fix: theme selector input

### PR DESCRIPTION
**Pull request by agazso**
This was trickier to get right than I originally thought. Some issues I ran into:
- In the original version the color chooser button background had the `theme.baseColor`, but I decided to remove this because it did not work well with the focused border of the button. Therefore the color input had to be made invisible inside a label so that it is invoked when the button is pressed. The button's padding also needed to be set to 0 explicitly.
- The `Input` value binding did not work because it made an internal copy of the value in a state variable (`inputValue`) but it never updated the bound value. I fixed it by simply using the `value`.
- There is a line that takes `theme.mode` and stores it in a derived value which is not used, but if that line is removed then the dropdown does not appear. It would be good to understand why and how to fix this.

### CODE REVIEW

File: src/lib/components/custom/theme-selector.svelte [hunk 1]
```diff
@@ -78,20 +67,38 @@
...
+		display: flex;
+		margin: 0;
+		outline: 0;
 		border: none;
-		border-radius: var(--border-radius);
+		border-radius: 0 var(--border-radius) var(--border-radius) 0;
+		padding: 0;
+		width: 100%;
+		height: 100%;
 	}
```
Comment (vojtechsimetka) [inline src/lib/components/custom/theme-selector.svelte:102]: These look pretty much the same for the webkit and mozilla version. Can it be unified and if needed the distinct css attributes added only the the relevant selector (here AFAIK it is just `display: flex` in mozilla)

```css
input[type='color']::-webkit-color-swatch,
input[type='color']::-moz-color-swatch {
		margin: 0;
		outline: 0;
		border: none;
		border-radius: 0 var(--border-radius) var(--border-radius) 0;
		padding: 0;
		width: 100%;
		height: 100%;
	}
	input[type='color']::-moz-color-swatch {
		display: flex;
	}
	```

File: src/lib/components/custom/theme-selector.svelte [hunk 2]
```diff
@@ -78,20 +67,38 @@
 		-webkit-appearance: none;
 		-moz-appearance: none;
 		appearance: none;
```
Comment (vojtechsimetka) [inline src/lib/components/custom/theme-selector.svelte:69]:
```suggestion
		appearance: none;
```

File: src/lib/components/custom/theme-selector.svelte [hunk 3]
```diff
@@ -3,21 +3,12 @@
...
-			? effectiveMode === 'dark'
-				? 'fill: var(--colors-base)'
-				: 'fill: var(--colors-top)'
-			: effectiveMode === 'dark'
-				? 'fill: var(--colors-top)'
-				: 'fill: var(--colors-base)',
-	)
+	// without this line the dropdown does not appear...
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	const mode = $derived(theme.mode)
```
Comment (vojtechsimetka) [inline src/lib/components/custom/theme-selector.svelte:11]: This is indeed very weird. Please make an issue and I'd be happy to figure out what is the problem.

**Review by vojtechsimetka — APPROVED** [review]

File: src/lib/components/custom/theme-selector.svelte (hunk 1 above)
Comment (agazso) [reply src/lib/components/custom/theme-selector.svelte:102]: BTW They can be identical, and also currently the color input is invisible so they are not strictly needed, unless we want to show the selected color in the button.

File: src/lib/components/custom/theme-selector.svelte (hunk 3 above)
Comment (agazso) [reply src/lib/components/custom/theme-selector.svelte:11]: Thanks, I opened an issue about it:
https://github.com/diete-design/diete.design/issues/139

Comment (david-gauquelin) [reply src/lib/components/custom/theme-selector.svelte:11]: About your first point, there was an issue in the design file, for some reason it was not using the actual color picker component. It's fixed now.

**Pull request merged**